src/include/sof/audio/codec_adapter/codec/cadence.h

```diff
@@ -36,11 +36,20 @@ struct cadence_api {
 	xa_codec_func_t *api;
 };
 
+enum default_stream_params_ids {
+	CADENCE_SAMPLE_RATE_ID = 0x01,
+	CADENCE_SAMPLE_WIDTH_ID = 0x02,
+	CADENCE_CHANNELS_COUNT_ID = 0x04
```

Comment:
Is this a bit mask from Cadence ? if so enum looks like the wrong choice.
Btw, @dbaluta how do you set this from the kernel ?

Reply:
No its not a bit mask, these are IDs for parameters setting. Unfortunately different cadence codes uses different IDs number for the same parameters, for instance one defines sample width as `2` while other as `2`. Hence we need to have different enums for that. 
PS: I should have assign decimal values as hexes are misleading I guess

Reply:
Yes, please use decimal and also a big comment that these are from Cadence and hence do not change the IDs.

Reply:
oh,  we may want to state in the comment, that this applies to codec X, Y and Z only (so everyone knows if not for all Cadence codecs)

Reply:
@lgirdwood we do not set this format right now from the kernel. But we really should! I am setting this from the topology.

Reply:
@mrajwa are this parameters default for your DAP codec? Is this mandatory for v1.8 release?

Reply:
@dbaluta lets add these params from compressed PCMs in v1.9 - I would also chat with @plbossart on compressed stream params.

Reply:
@lgirdwood agree.